### ISSUE TITLE
Why Is APSIM Consuming Excessive Resources on Google Cloud's Linux Environment?

### ISSUE THREAD

**Issue by overg123** [APSIMInitiative/ApsimX#9228]
### What happened?

Hi,

I am running APSIM version 7.2.8.1 on a Linux system in Google Cloud. My setup includes optimized cloud computing with 16 cores and 64GB of memory. Despite this, running APSIM with the attached file (and in many other scenarios) takes over 15 minutes.

During this time, I observed that APSIM uses all 16 cores at 100% utilization. This seems highly unreasonable. Could there be an issue with the APSIM files, the weather files, or is there another way to optimize its operation?

[Soy_Minnesota.zip](https://github.com/user-attachments/files/16508535/Soy_Minnesota.zip)

### What Operating System are you on?

Windows

**Comment by hol353** [comment]
APSIM will use all available CPU so using all 16 cores is ok. I notice in your simulations that you are outputting daily. This makes APSIM very slow as there are a lot of numbers to write to the .db file. Try removing this and see if that runs quicker. Report has the ability the aggregate (sum, mean etc) over a period (https://builds.apsim.info/api/nextgen/docs/Report.pdf) which you can use instead of writting daily values.

**Comment by par456** [comment]
I just ran this myself on my local machine and the 5.1K simulations only took about 3 minutes to run, which is what I would expect of an experiment of this size. I then ran apsim using wsl to test it on linux and while the simulations ran in about 3 minutes again, it then had a queue of 270K operations that still needed to be written to the database, which took over 10 minutes to finish. And throwing more cores at it won't solve that problem.

That seems to suggest that there is a bottleneck in the linux build where it is not able to write to the datastore as quickly. Dean's suggestion above will help (not writing all variables every day to the datastore), however there does seem to be a bug here that needs to be addressed on the linux build.
